src/boat/private/BoatConfig.nim

```diff
@@ -0,0 +1,100 @@
+##
+## Config
+## ======
+## extendable interface for creating, parsing and saving boat configs
+
+##[
+## TLDR
+- come back later
+
+todos
+-----
+- cmd to parse arbitrarily, e.g. boat conf load ./some/dir
+- cmd to see current captains: e.g. boat conf list -> X, Y, Z
+]##
+
+from ../../../bdd import tddError
+
+import std/[
+  json,
+  os,
+  strutils,
+  ]
+
+import
+  BoatConstants,
+  BoatErrors,
+  BoatConfigType,
+  FileManager
+
+# consumers can retrieve the parsed Config and path on disk
+# but only internal functions should be able to set it
+export BoatConfigType, parsed, parsedPath
+
+var captainsLog* {.global.} = %* {} ## \
+  ## captains log is the world
+
+
+proc parseLocalManifest*(self: BoatConfig, path = self.use): bool =
+  ## parse self.use to self.parsed
+  ## prefer calling self.load or self.reload for validation
+  self.parsed = loadConfig path
+  result = true
+
+proc localManifestIsValid*(self: BoatConfig, path: string = self.use): bool =
+  ## throws if manifest not found, cant be read, or errors during parsing
+  let pathInfo = path.getFileInfo
+  result = case pathInfo.kind
+    of pcFile, pcLinkToFile:
+      if fpUserRead notin pathInfo.permissions: raise filePermissionError
+      elif not path.endsWith manifestName: raise manifestNameError
+      elif not self.parseLocalManifest path: raise configParseError
```

Comment:
this should just be self.parsed = path
check BoatConfigType parsed= which should set self.parsed = parsecfg.loadConfig path